portal-api/src/v1/gef/controllers/clone-gef-deal.controller.js

```diff
@@ -1,182 +1,46 @@
-
-  // get the current GEF deal
-  const existingDeal = await collection.findOne({ _id: { $eq: ObjectId(dealId) }, 'bank.id': { $eq: bank.id } });
-  if (existingDeal) {
-    const clonedDeal = existingDeal;
-    const eligibility = await getLatestEligibilityCriteria();
-
-    // delete unused properties
-    unusedProperties.forEach((property) => {
-      if (clonedDeal[property]) {
-        delete clonedDeal[property];
-      }
-    });
-
-    clonedDeal.createdAt = Date.now();
-    clonedDeal.updatedAt = Date.now();
-    clonedDeal.facilitiesUpdated = Date.now();
-    clonedDeal.eligibility = eligibility;
-    clonedDeal.eligibility.updatedAt = Date.now();
-    clonedDeal.status = CONSTANTS.DEAL.DEAL_STATUS.DRAFT;
-    clonedDeal.submissionType = null;
-    clonedDeal.submissionDate = null;
-    clonedDeal.submissionCount = 0;
-    clonedDeal.bankInternalRefName = bankInternalRefName;
-    clonedDeal.additionalRefName = additionalRefName;
-    clonedDeal.maker = maker;
-    clonedDeal.bank = bank;
-    clonedDeal.ukefDealId = null;
-    clonedDeal.checkerId = null;
-    clonedDeal.editedBy = [userId];
-    clonedDeal.exporter = cloneExporter(clonedDeal.exporter);
-    clonedDeal.portalActivities = [];
-    clonedDeal.supportingInformation = {};
-    clonedDeal.clonedDealId = dealId;
-    clonedDeal.auditRecord = generateAuditDatabaseRecordFromAuditDetails(auditDetails);
-
-    // insert the cloned deal in the database
-    const createdApplication = await collection.insertOne(clonedDeal);
-    const newDealId = createdApplication.insertedId;
-
-    // return the ID for the newly inserted deal
-    return { newDealId: newDealId.toHexString(), status: 200 };
-  }
-  return { status: 404 };
-};
 
 exports.clone = async (req, res) => {
   const {
     body: { dealId: existingDealId, bankInternalRefName, additionalRefName, userId, bank },
   } = req;
 
-  const validateErrs = validateApplicationReferences(req.body);
+  try {
+    const validationErrors = validateApplicationReferences(req.body);
+
+    if (validationErrors) {
+      return res.status(422).send(validationErrors);
+    }
+
+    const auditDetails = generatePortalAuditDetails(req.user._id);
+    // clone GEF deal
+    const response = await cloneDealToLatestVersion({
+      dealId: existingDealId,
+      bankInternalRefName,
+      additionalRefName,
+      maker: req.user,
+      userId,
+      bank,
+      auditDetails,
+    });
 
-  if (validateErrs) {
-    return res.status(422).send(validateErrs);
-  }
+    const { insertedId } = response;
+    // clone the corresponding facilities
+    await cloneFacilities(existingDealId, insertedId, auditDetails);
 
-  const auditDetails = generatePortalAuditDetails(req.user._id);
-  // clone GEF deal
-  const response = await cloneDeal({
-    dealId: existingDealId,
-    bankInternalRefName,
-    additionalRefName,
-    maker: req.user,
-    userId,
-    bank,
-    auditDetails,
-  });
+    return res.status(200).send({ dealId: insertedId });
```

Comment:
could we use HttpStatusCode here? for the 200 and 422?

Reply:
@Zainzzkk are you happy to resolve this?